src/components/GoalParentDropdown/GoalParentDropdown.tsx

```diff
@@ -11,24 +11,40 @@ import { ModalEvent, dispatchModalEvent } from '../../utils/dispatchModal';
 import s from './GoalParentDropdown.module.css';
 import { tr } from './GoalParentDropdown.i18n';
 
-interface GoalParentDropdownProps {
+interface GoalParentValue {
+    id: string;
+    title: string;
+}
+
+type GoalParentDropdownProps = {
     error?: ComponentProps<typeof DropdownTrigger>['error'];
     label?: ComponentProps<typeof DropdownTrigger>['label'];
-    value?: { id: string; title: string };
+    value?: GoalParentValue | GoalParentValue[];
     query?: string;
     className?: string;
     placeholder?: string;
     disabled?: boolean;
-    onChange?: (project: { id: string; title: string }) => void;
-}
+    placement?: ComponentProps<typeof DropdownPanel>['placement'];
+    onClose?: () => void;
+} & (
+    | {
```

Comment:
U can export some type helpers with generic for in place type define. Smth like: `& DropdownGuardedProps<GoalParentValue>`